geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/sortedset/ZRemRangeByLexDUnitTest.java

```diff
@@ -0,0 +1,280 @@
+    int locatorPort = locator.getPort();
+    MemberVM server1 = clusterStartUp.startRedisVM(1, locatorPort);
+    MemberVM server2 = clusterStartUp.startRedisVM(2, locatorPort);
+    servers = new ArrayList<>();
+    servers.add(server1);
+    servers.add(server2);
+
+    int redisServerPort1 = clusterStartUp.getRedisPort(1);
+    int redisServerPort2 = clusterStartUp.getRedisPort(2);
+
+    jedis1 =
+        new JedisCluster(new HostAndPort(BIND_ADDRESS, redisServerPort1), REDIS_CLIENT_TIMEOUT);
+    jedis2 =
+        new JedisCluster(new HostAndPort(BIND_ADDRESS, redisServerPort2), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    jedis1.close();
+  }
+
+  @Test
+  public void ZRemRangeByLex_removesMembersInRangeFromBothServers() {
+    Map<String, Double> memberScoreMap = makeMemberScoreMap(setSize);
+    jedis1.zadd(sortedSetKey, memberScoreMap);
+    verifyDataExists(memberScoreMap);
+
+    List<String> expected = new ArrayList<>(memberScoreMap.keySet().stream().sorted()
+        .collect(Collectors.toList())).subList(4, setSize);
+
+    assertThat(jedis1.zrank(sortedSetKey, "vvvvv")).isEqualTo(4L);
+    assertThat(jedis2.zrank(sortedSetKey, "vvvvv")).isEqualTo(4L);
+
+    assertThat(jedis1.zremrangeByLex(sortedSetKey, "[" + "v", "[" + "vvvv")).isEqualTo(4);
+
+    assertThat(jedis1.zrank(sortedSetKey, "vvvvv")).isEqualTo(0L);
+    assertThat(jedis2.zrank(sortedSetKey, "vvvvv")).isEqualTo(0L);
+
+    assertThat(jedis1.zrange(sortedSetKey, 0, -1)).hasSameElementsAs(expected);
+    assertThat(jedis2.zrange(sortedSetKey, 0, -1)).hasSameElementsAs(expected);
+  }
+
+  @Test
+  public void ZRemRangeByLex_concurrentlyRemovesMembersInRangeFromBothServers() {
+    Map<String, Double> memberScoreMap = makeMemberScoreMap(setSize);
+    jedis1.zadd(sortedSetKey, memberScoreMap);
+    verifyDataExists(memberScoreMap);
+
+    assertThat(jedis1.zrank(sortedSetKey, "vvvvv")).isEqualTo(4L);
+    assertThat(jedis2.zrank(sortedSetKey, "vvvvv")).isEqualTo(4L);
+
+    AtomicInteger totalRemoved = new AtomicInteger();
+    new ConcurrentLoopingThreads(setSize,
+        (i) -> doZRemRangeByLexOnMembers(i, totalRemoved),
+        (i) -> doZRemRangeByLexOnMembersInDifferentOrder(i, totalRemoved)).run();
+
+    assertThat(jedis1.exists(sortedSetKey)).isFalse();
+    assertThat(jedis2.exists(sortedSetKey)).isFalse();
+
+    assertThat(totalRemoved.get()).isEqualTo(setSize);
+  }
+
+  @Test
+  public void zRemRangeByLexRemovesMembersFromSortedSetAfterPrimaryShutsDown() {
+    Map<String, Double> memberScoreMap = makeMemberScoreMap(setSize);
+    jedis1.zadd(sortedSetKey, memberScoreMap);
+    verifyDataExists(memberScoreMap);
+
+    stopNodeWithPrimaryBucketOfTheKey(false);
+
+    doZRemRangeByLexWithRetries(memberScoreMap);
+
+    verifyDataDoesNotExist(memberScoreMap);
+    assertThat(jedis1.exists(sortedSetKey)).isFalse();
+    assertThat(jedis2.exists(sortedSetKey)).isFalse();
+  }
+
+  @Test
+  @Ignore("Fails due to GEODE-9310")
```

Comment:
This should be safe to un-ignore if a third server is added to the test. A working implementation of what this test is trying to do can be found in `ZRemRangeByScoreDUnitTest`.